### PR TITLE
TASK: Declare fusion core classes as internal

### PR DESCRIPTION
**Pull request by mhsdesign**
- Marks newer super internal low level Fusion parsing details as internal
  - This structure was introduced with the new Fusion Parser in Neos 8 https://github.com/neos/neos-development-collection/pull/3497
- Declares Fusion's Core as @internal
  - For interacting with Fusion from the outside a FusionView should be used

Custom Fusion object implementations might still rely on manipulating/calling the Runtime directly,
if the abstractions in the AbstractFusionObject doesn't suffice.
For that use case we should implement a more powerful and clean abstraction:
https://github.com/neos/neos-development-collection/issues/4910

Also i cleaned up the new Fusion parsers AST objects to use modern php 8.2 that wasnt available two years ago :D

**Upgrade instructions**

**Review instructions**

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.Fusion/Classes/Core/Parser.php [hunk 1]
```diff
@@ -25,7 +25,8 @@
 /**
  * The Fusion Parser
  *
- * @api
+ * @internal The Fusion parsing is considered internal.
+ *           For interacting with Fusion from the outside a FusionView should be used.
```
Comment (mhsdesign) [inline Neos.Fusion/Classes/Core/Parser.php:29]: the parser being api was introduced via this commit: https://github.com/neos/neos-development-collection/commit/fde5445127f83a3ca8a1d2c2cc6ea3fc6fcc07e8

But if the runtime should not be api (which it should not) one wont need the parser to be api :)

TODO: Fix api annotation in `parseFromSource`

**Comment by kitsunet** [comment]
The Views (and their interface) are a pure Flow concept. I think we talked about the idea to make fusion standaloneable (which is probably feasible over time) but then Runtime would have to be the primary API? Or do we disregard this idea entirely? 

The whole parsing part def internal.

**Comment by mhsdesign** [comment]
> I think we talked about the idea to make fusion standaloneable (which is probably feasible over time) but then Runtime would have to be the primary API? Or do we disregard this idea entirely?

Yes i _do_ think so but then the runtime would need an overhauled api in general and not 3 entry points `renderResponse` `render` and `evaluate`.
When we _do_ make fusion standalone we would be free to design the api as we want, thats why its important to make sure its understood as internal for now.

https://github.com/neos/neos-development-collection/issues/4910 already discusses what fusion objects should see from the runtime, as they might need some kind of direct access.

**Review by kitsunet — APPROVED** [review]

File: Neos.Fusion/Classes/Core/ObjectTreeParser/Exception/ParserException.php [hunk 2]
```diff
@@ -18,6 +18,7 @@
 
 /**
  * 'Fluent' exception for the Fusion Parser.
+ * @internal although ${exception.headingMessagePart} ${exception.asciiPreviewMessagePart} and ${exception.helperMessagePart} are used outside for custom error display
```
Comment (grebaldi) [inline Neos.Fusion/Classes/Core/ObjectTreeParser/Exception/ParserException.php]: Then it ain't internal, I'd argue. But it's perfectly fine (even necessary) for parser exceptions to be API imho.

**Review by grebaldi — COMMENTED** [review]
Hi @mhsdesign,

:+1: by reading. Just one note on one @internal exception that may have use outside Neos.Fusion.

File: Neos.Fusion/Classes/Core/ObjectTreeParser/Exception/ParserException.php (hunk 2 above)
Comment (kitsunet) [reply Neos.Fusion/Classes/Core/ObjectTreeParser/Exception/ParserException.php]: If the runtime and parser are internal you shouldn't have any reason to try/catch this in userland code I guess?

Comment (mhsdesign) [inline Neos.Fusion/Classes/Core/ObjectTreeParser/Exception/ParserException.php]:
```suggestion
 * @internal
```

**Review by mficzel — APPROVED** [review]

**Pull request merged**